packages/jest-runtime/src/transform.js

```diff
@@ -159,6 +161,18 @@ const shouldInstrument = (filename: Path, config: Config): boolean => {
   }
 
   if (
+    !config.collectCoverageOnlyFrom && // still cover if `only` is specified
+    config.collectCoverageFrom &&
+    !multimatch(
```

Comment:
I haven't used this myself, but may be worth checking out for glob performance: https://github.com/jonschlinkert/micromatch